### PR TITLE
SUP-1401: Cluster Queue and Cluster Agent Token resource retries

### PR DESCRIPTION
**Pull request by james2791**
feat (Timeouts/Retries): Add timeouts (provider config set) and CRUD retries to the Cluster queue and Cluster agent token resources.

## PR checklist:
- [x] `docs/` updated (provider docs updated previously)
- [x] tests added (t.Run and cluster randomness conversion)
- [x] an example added to `examples/` (provider example updated with `timeouts` block)
- [x] `CHANGELOG.md` updated with pending release information

Introduced in this PR - all Cluster queue/token tests have been converted to `t.Run`, and are also changed in that a seperate cluster is created for each individual test (vs previously being fixed for both test types) of the name `Test cluster <10 chars>`. `CheckDestroy` for each test verifies that the cluster is torn down once the create/update/import is verified (eventual node shift to come as per the older PRs [here](https://github.com/buildkite/terraform-provider-buildkite/pull/321) and [here](https://github.com/buildkite/terraform-provider-buildkite/pull/322))

As part of this PR - I've moved the `retryContextError` function from the pipeline resource to `util.go` - which the other resources/datasources could use. All under the `buildkite` package so not needing an export unless that changes, very unlikely.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,7 +44,7 @@ functionality to [lifecycles](https://developer.hashicorp.com/terraform/language
 
 - Move archive pipeline config to provider [[PR #354](https://github.com/buildkite/terraform-provider-buildkite/pull/354)] @jradtilbrook
 - SUP-1076 Convert testing to framework [[PR #361](https://github.com/buildkite/terraform-provider-buildkite/pull/361)] @jradtilbrook
-    - SUP-1076 Move Cluster tests to use t.Run [[PR #365](https://github.com/buildkite/terraform-provider-buildkite/pull/365)] @mcncl
+- SUP-1076 Move Cluster tests to use t.Run [[PR #365](https://github.com/buildkite/terraform-provider-buildkite/pull/365)] @mcncl
```
Comment (mcncl) [inline CHANGELOG.md:47]: This indenting is intentional; it puts the Cluster test changes as a part of the above bullet point.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,8 @@ All notable changes to this project will be documented in this file.
 - SUP-1322: Team member resource retries [[PR #381](https://github.com/buildkite/terraform-provider-buildkite/pull/381)] @james2791
 - SUP-1402: Agent token resource retries [[PR #382](https://github.com/buildkite/terraform-provider-buildkite/pull/382)] @james2791
 - SUP-1399: Add retry to pipeline team resource [[PR #384](https://github.com/buildkite/terraform-provider-buildkite/pull/384)] @lizrabuya
+- SUP-1361: Add timeouts to pipeline resource api[[PR #385](https://github.com/buildkite/terraform-provider-buildkite/pull/385)] @lizrabuya
```
Comment (jradtilbrook) [inline CHANGELOG.md:16]: Duplicate line
```suggestion
```

**Review by jradtilbrook — CHANGES_REQUESTED** [review]
Just one minor thing

**Review by mcncl — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (james2791) [reply CHANGELOG.md]: Ta - corrected in the latest

File: CHANGELOG.md (hunk 1 above)
Comment (james2791) [reply CHANGELOG.md]: Ah nice - Ive rolled it back with the latest

**Review by jradtilbrook — APPROVED** [review]

**Review by mcncl — APPROVED** [review]

**Pull request merged**